src/internal/hooks/use-performance-marks/index.ts

```diff
@@ -1,20 +1,44 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useEffect } from 'react';
-import { useUniqueId } from './use-unique-id';
-import { useEffectOnUpdate } from './use-effect-on-update';
-import { useModalContext } from '../context/modal-context';
+import { useEffect, useRef } from 'react';
+import { useIdFallback } from '../use-unique-id';
+import { useEffectOnUpdate } from '../use-effect-on-update';
+import { useModalContext } from '../../context/modal-context';
 
+/*
+This hook allows setting an HTML attribute after the first render, without rerendering the component.
+*/
+function useAttributeWithoutRerendering(elementRef: React.RefObject<HTMLElement>, value: string) {
+  const attributeName = 'data-analytics-performance-mark';
+
+  const attributeValueRef = useRef<string | undefined>();
+
+  useEffect(() => {
+    attributeValueRef.current = value;
+    elementRef.current?.setAttribute(attributeName, value);
```

Comment:
How about assigning `__awsuiMetadata__` property here instead of an attribute? Would it make it easier to maintain?

Reply:
Actually, if we eliminate useEffect completely (https://github.com/cloudscape-design/components/pull/2345#discussion_r1631105798), the attribute is fine

Reply:
If we assigned it to a property instead of a HTML attribute, it would be more expensive to query the DOM for the element. With the attribute we can run `querySelector`, but with a property we would probably have to loop through every DOM element until we find the one with the property set to the right value.

Reply:
"expensive to query" needs some supportive data. For instance, [TreeWalker DOM API](https://developer.mozilla.org/en-US/docs/Web/API/Treewalker) exists.

I can approve the version with attribute, just saying that some assumptions need validation

Reply:
Interesting, I did not know about the TreeWalker API! I've created a small benchmark (on jsbench.me) using the HTML from [our Table demo](https://cloudscape.design/examples/react/table.html), and it confirms that `querySelector` is much faster (or that I'm using the TreeWalker inefficiently)

![benchmark](https://github.com/cloudscape-design/components/assets/9086585/634db26a-55e0-46fd-909f-06fb0ee08624)

Reply:
1. Even if it is slower, it is unclear if it matters for this use-case (depends how often this lookup is called)
2. We can also fine tune the benchmark, but this is not important for the PR, the attributes can stay